boards/arduino/arduino_giga_r1/Kconfig.arduino_giga_r1

```
@@ -0,0 +1,10 @@
+# Copyright 2023 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_ARDUINO_GIGA_R1_STM32H747XX_M7
+	bool
+	select SOC_STM32H747XX_M7
+
+config BOARD_ARDUINO_GIGA_R1_STM32H747XX_M4
+	bool
+	select SOC_STM32H747XX_M4
```

> Is that the expected way to proceed ?

No, please do like this:
```suggestion
config BOARD_ARDUINO_GIGA_R1
	select SOC_STM32H747XX_M7 if BOARD_ARDUINO_GIGA_R1_STM32H747XX_M7
	select SOC_STM32H747XX_M4 if BOARD_ARDUINO_GIGA_R1_STM32H747XX_M4
```

this way it's ensured that Kconfig will fail with undefined type for symbol if board name is mismatching / renamed, as well as CI will fail if the `select <SOC>` or `if BOARD_<full_identifier>` are wrong / renamed.